### ISSUE TITLE
[8.0][auth_admin_passkey] Error when logging in with same password

### ISSUE THREAD

**Issue by thomaspaulb** [OCA/server-tools#699]
Logging in with user who has same password as admin user brings an error:

```
  File "/opt/odoo8/openerp/service/common.py", line 38, in exp_authenticate
    return res_users.authenticate(db, login, password, user_agent_env)
  File "/home/terrence/odoo-custom-addons/sunflower/auth_admin_passkey/models/res_users.py", line 100, in authenticate
    self._send_email_same_password(cr, login)
  File "/opt/odoo8/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
TypeError: _send_email_same_password() takes exactly 2 arguments (3 given)
```

**Comment by alexis-via** [comment]
I confirm this bug on v8 with up-to-date code.

**Comment by legalsylvain** [comment]
This commit should have introduced the bug.
https://github.com/OCA/server-tools/commit/58ec085e4c3b0ccd196a94c226ca006c2e4c43f9

 I did not know @ api.cr decorator. 
I guess that this changes should fix the problem : 

        @api.cr
        def _send_email_same_password(self, login_user):

replaced by 

        @api.model
        def _send_email_same_password(self, login_user):

and 

                if not same_password:
                    self._send_email_passkey(cr, user_id, user_agent_env)
                else:
                    self._send_email_same_password(cr, login)

by 

                if not same_password:
                    self._send_email_passkey(cr, user_id, user_agent_env)
                else:
                    self._send_email_same_password(cr, user_id, login)



CC : @rvalyi, @pedrobaeza.

**Comment by alexis-via** [comment]
yes, the temporary workaround is "git co 464ccb6b59036ad38a55c649e1d2bc37e701481d" :)

**Comment by bealdav** [comment]
Really it's the workaround ?
really weird :)

**Comment by legalsylvain** [comment]
@rvalyi, could you take a look on this bug ? 
kind regards.

**Comment by alexis-via** [comment]
I made a fix in this PR https://github.com/OCA/server-tools/pull/752/files

**Comment by alexis-via** [comment]
There is also another bug in this module: when you uncheck the 2 options to send mail on the configuration page, you get this error on authentification:

```
2017-02-28 15:12:11,589 12095 ERROR o8_test1 openerp.http: unexpected EOF while parsing
('', 0, 0, '')
Traceback (most recent call last):
  File "/home/alexis/new_boite/dev/odoo8/openerp/http.py", line 115, in dispatch_rpc
    result = dispatch(method, params)
  File "/home/alexis/new_boite/dev/odoo8/openerp/service/common.py", line 26, in dispatch
    return fn(*params)
  File "/home/alexis/new_boite/dev/odoo8/openerp/service/common.py", line 38, in exp_authenticate
    return res_users.authenticate(db, login, password, user_agent_env)
  File "/home/alexis/new_boite/dev/server-tools8/auth_admin_passkey/models/res_users.py", line 102, in authenticate
    self._send_email_passkey(cr, user_id, user_agent_env)
  File "/home/alexis/new_boite/dev/odoo8/openerp/api.py", line 268, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/alexis/new_boite/dev/odoo8/openerp/api.py", line 372, in old_api
    result = method(recs, *args, **kwargs)
  File "/home/alexis/new_boite/dev/server-tools8/auth_admin_passkey/models/res_users.py", line 34, in _send_email_passkey
    'True'))
  File "/home/alexis/new_boite/dev/odoo8/openerp/tools/safe_eval.py", line 312, in safe_eval
    c = test_expr(expr, _SAFE_OPCODES, mode=mode)
  File "/home/alexis/new_boite/dev/odoo8/openerp/tools/safe_eval.py", line 167, in test_expr
    code_obj = compile(expr, "", mode)
  File "<string>", line 0
    
    ^
SyntaxError: unexpected EOF while parsing
> /home/alexis/new_boite/dev/odoo8/openerp/tools/safe_eval.py(167)test_expr()
-> code_obj = compile(expr, "", mode)
```

**Comment by legalsylvain** [comment]
@thomaspaulb, @rvalyi, @bealdav : could you review the fix of Alexis ? #752 

@alexis-via : I guess it's a unrelated bug. Nothing to do with this one. Creating a new issue : #754
